src/test/ui/issues/issue-66856.rs

```diff
@@ -0,0 +1,19 @@
+// build-pass
```

Comment:
Let's convert this to a mir-opt test instead of a build-pass test to avoid relying on what LLVM does and doesn't do.

Reply:
Here's what I had (in `src/test/mir-opt/simplify-arm-identity.rs`):
```rust
enum Src {
    Foo(u8),
    Bar,
}

enum Dst {
    Foo(u8),
}

fn main() {
    let e: Src = Src::Foo(0);
    let _: Dst = match e {
        Src::Foo(x) => Dst::Foo(x),
        Src::Bar => Dst::Foo(0),
    };
}

// END RUST SOURCE
// START rustc.main.SimplifyArmIdentity.before.mir
// fn main() -> () {
//     ...
//     bb0: {
//         StorageLive(_1);
//         ((_1 as Foo).0: u8) = const 0u8;
//         discriminant(_1) = 0;
//         StorageLive(_2);
//         _3 = discriminant(_1);
//         switchInt(move _3) -> [0isize: bb3, 1isize: bb1, otherwise: bb2];
//     }
//     bb1: {
//         ((_2 as Foo).0: u8) = const 0u8;
//         discriminant(_2) = 0;
//         goto -> bb4;
//     }
//     ...
//     bb3: {
//         _4 = ((_1 as Foo).0: u8);
//         ((_2 as Foo).0: u8) = move _4;
//         discriminant(_2) = 0;
//         goto -> bb4;
//     }
//     ...
// }
// END rustc.main.SimplifyArmIdentity.before.mir
// START rustc.main.SimplifyArmIdentity.after.mir
// fn main() -> () {
//     ...
//     bb0: {
//         StorageLive(_1);
//         ((_1 as Foo).0: u8) = const 0u8;
//         discriminant(_1) = 0;
//         StorageLive(_2);
//         _3 = discriminant(_1);
//         switchInt(move _3) -> [0isize: bb3, 1isize: bb1, otherwise: bb2];
//     }
//     bb1: {
//         ((_2 as Foo).0: u8) = const 0u8;
//         discriminant(_2) = 0;
//         goto -> bb4;
//     }
//     ...
//     bb3: {
//         _4 = ((_1 as Foo).0: u8);
//         ((_2 as Foo).0: u8) = move _4;
//         discriminant(_2) = 0;
//         goto -> bb4;
//     }
//     ...
// }
// END rustc.main.SimplifyArmIdentity.after.mir
```